src/util/thread.h

```diff
@@ -26,6 +26,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "../threading/mutex_auto_lock.h"
 #include "porting.h"
 #include "log.h"
+#include "container.h" // MutexedQueue
```

Comment:
I mean, if you ctrl+f that file, MutexedQueue is required there. It's just that the compiler doesn't care until you actually try to use the classes in util/thread.h as they require template parameters.